uitools/cpp/Esri/ArcGISRuntime/Toolkit/UtilityNetworkTraceController.cpp

```diff
@@ -0,0 +1,1026 @@
+                                                                         QColor(0, 0, 255, 126),
+                                                                         20,
+                                                                         this);
+          auto graphic = new Graphic(multipoint,
+                                     resultPointSymbol,
+                                     this);
+          hasGeometryTraceResult = true;
+          m_resultsGraphicsOverlay->graphics()->append(graphic);
+        }
+
+        auto polyline = geometryTraceResult->polyline();
+        if (!polyline.isEmpty())
+        {
+          SimpleLineSymbol* resultPointSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Dot,
+                                                                     QColor(0, 0, 255, 126),
+                                                                     5,
+                                                                     this);
+          auto graphic = new Graphic(polyline,
+                                     resultPointSymbol,
+                                     this);
+          hasGeometryTraceResult = true;
+          m_resultsGraphicsOverlay->graphics()->append(graphic);
+        }
+
+        auto polygon = geometryTraceResult->polygon();
+        if (!polygon.isEmpty())
+        {
+          SimpleFillSymbol* resultPointSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle::ForwardDiagonal,
+                                                                     QColor(0, 0, 255, 126),
+                                                                     new SimpleLineSymbol(SimpleLineSymbolStyle::Solid,
+                                                                                          QColor(0, 0, 255, 126),
+                                                                                          2,
+                                                                                          this),
+                                                                     this);
+          auto graphic = new Graphic(polygon,
+                                     resultPointSymbol,
+                                     this);
+          hasGeometryTraceResult = true;
+          m_resultsGraphicsOverlay->graphics()->append(graphic);
+        }
+
+        break;
+      }
+      default:
+      {
+        qDebug() << "Trace completed without usable object type";
+        break;
+      }
+    }
+
+  }
+
+  if (!allElements.isEmpty())
+  {
+    // calling async featuresForElements with elements
+    m_selectedUtilityNetwork->featuresForElements(allElements);
+  }
+  else
+  {
+    // when the above async completes, the run trace in-progress will be set to false
+    setIsTraceInProgress(false);
+  }
+
+  if (hasGeometryTraceResult)
+  {
+    auto mapView = qobject_cast<MapViewToolkit*>(m_geoView);
+    mapView->graphicsOverlays()->append(m_resultsGraphicsOverlay);
```

Comment:
this feels like you will append the graphics overlay again and again, whenever you complete a trace result. I think instead you should add it just once (maybe after the first trace result) and then clear/delete the graphics from it when you refresh etc.

Reply:
Indeed! Done